File: 1508-range-sum-of-sorted-subarray-sums/1508-range-sum-of-sorted-subarray-sums.py
class Solution:
    def rangeSum(self, nums: List[int], n: int, left: int, right: int) -> int:
        subarr = []
        
        for i in range(n):
            partial = 0
            for j in range(i, n):
                partial = (partial + nums[j]) % (1_000_000_007)
                subarr.append(partial)
        
        subarr.sort()
        
        return sum(subarr[left - 1: right]) % (1_000_000_007)

File: 1508-range-sum-of-sorted-subarray-sums/1508-range-sum-of-sorted-subarray-sums.cpp
typedef long long ll;

class Solution {
public:
    int rangeSum(vector<int>& nums, int n, int left, int right) {
        vector<ll> sumSubarr(n * (n + 1) >> 1);
        int k = 0, rem = 1e9 + 7;
        
        for (int i = 0; i < n; ++i){
            ll partial = 0;
            for (int j = i; j < n; ++j){
                partial = (partial + nums[j]) % rem;
                sumSubarr[k++] = partial;
            }
        }
        sort(sumSubarr.begin(), sumSubarr.end());
        
        return accumulate(sumSubarr.begin() + left - 1, sumSubarr.begin() + right, 0LL) % rem;
    }
};